### ISSUE TITLE
StateError: Bad state: The client closed with pending request "streamListen"

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#66514]
Some of these are happening on run, some on attach, and some from IDEs. There are about 16 occurrences on the latest dev, which is concerning because it has only been available for a day or so

```
  | at new Client.withoutJson.<anonymous closure> | (client.dart:70)
-- | -- | --
  | at _rootRun | (zone.dart:1182)
  | at _CustomZone.run | (zone.dart:1093)
  | at _FutureListener.handleWhenComplete | (future_impl.dart:167)
  | at Future._propagateToListeners.handleWhenCompleteCallback | (future_impl.dart:666)
  | at Future._propagateToListeners | (future_impl.dart:722)
  | at Future._propagateToListeners | (future_impl.dart:621)
  | at Future._completeWithValue | (future_impl.dart:529)
  | at Future._asyncCompleteWithValue.<anonymous closure> | (future_impl.dart:567)
  | at _rootRun | (zone.dart:1190)
  | at _CustomZone.run | (zone.dart:1093)
  | at _CustomZone.runGuarded | (zone.dart:997)
  | at _CustomZone.bindCallbackGuarded.<anonymous closure> | (zone.dart:1037)
  | at _microtaskLoop | (schedule_microtask.dart:41)
  | at _startMicrotaskLoop | (schedule_microtask.dart:50)
  | at _runPendingImmediateCallback | (isolate_patch.dart:118)
  | at _RawReceivePortImpl._handleMessage | (isolate_patch.dart:169)
```

**Comment by jonahwilliams** [comment]
See also: https://github.com/flutter/flutter/issues/66554

**Comment by jonahwilliams** [comment]
@bkonyi

**Comment by bkonyi** [comment]
Do we happen to have a reproduction case for this? It's likely a connection error (improving the error is on my plate this week).

**Comment by jmagman** [comment]
Duplicate at https://github.com/flutter/flutter/issues/66180.

**Comment by bkonyi** [comment]
Have a CL [up for review](https://dart-review.googlesource.com/c/sdk/+/164601) that should cleanup most of the error handling in `package:dds`. Will require some changes in the tool as well but I have something working locally.

**Comment by abaturyna** [comment]
i am having the same issue ! my dependencies are : 

sqflite: ^1.3.1+1
  csv: ^4.0.3
  flutter_exif_rotation: ^0.2.7
  cupertino_icons: ^1.0.0
  image_picker: ^0.6.7+10
  fuzzy: ^0.2.3
  gallery_saver: ^2.0.1
  string_splitter: ^0.1.0+5
  image: ^2.1.17
  admob_flutter: ^1.0.0
  isolate_handler: ^0.3.1
  rate_my_app: ^0.7.2
  launch_review: ^2.0.0
  purchases_flutter: ^1.3.1
  firebase_ml_vision: ^0.9.6+2
  number_display: ^2.3.0
  custom_radio_button: ^0.0.3
  flutter_swiper: ^1.1.6
  flutter_spinkit: ^4.1.2+1
  circular_check_box: ^1.0.4
  font_awesome_flutter: ^8.8.1
  url_launcher: ^5.7.0
  intl: ^0.16.1

**Comment by abaturyna** [comment]
my stacktrace + flutter doctor is : 
Flutter crash report.
Please report a bug at https://github.com/flutter/flutter/issues.

## command

flutter run --verbose

## exception

StateError: Bad state: The client closed with pending request "streamListen".

```
#0      new Client.withoutJson.<anonymous closure> (package:json_rpc_2/src/client.dart:70:24)
#1      _rootRun (dart:async/zone.dart:1182:47)
#2      _CustomZone.run (dart:async/zone.dart:1093:19)
#3      _FutureListener.handleWhenComplete (dart:async/future_impl.dart:167:18)
#4      Future._propagateToListeners.handleWhenCompleteCallback (dart:async/future_impl.dart:666:39)
#5      Future._propagateToListeners (dart:async/future_impl.dart:722:37)
#6      Future._propagateToListeners (dart:async/future_impl.dart:621:9)
#7      Future._completeWithValue (dart:async/future_impl.dart:529:5)
#8      Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
#9      _rootRun (dart:async/zone.dart:1190:13)
#10     _CustomZone.run (dart:async/zone.dart:1093:19)
#11     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
#12     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
#13     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#14     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#15     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#16     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
```

## flutter doctor

```
[✓] Flutter (Channel dev, 1.23.0-4.0.pre, on Mac OS X 10.15.7 19H2 x86_64, locale ru-BY)
    • Flutter version 1.23.0-4.0.pre at /Users/nastya/Dropbox/BUSINESS/Soft/flutter 1.2.13
    • Framework revision 83dd176777 (5 дней назад), 2020-09-22 12:04:44 -0700
    • Engine revision 2abe69c608
    • Dart version 2.10.0 (build 2.10.0-136.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/nastya/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 45.1.3
    • Dart plugin version 201.7223.97

[✓] Connected device (3 available)
    • nastenka’s iPhone (mobile) • 00008030-000A113802E2802E • ios            • iOS 14.0.1
    • Web Server (web)           • web-server                • web-javascript • Flutter Tools
    • Chrome (web)               • chrome                    • web-javascript • Google Chrome 85.0.4183.121

• No issues found!

**Comment by jonahwilliams** [comment]
Based on the number of new reports I am increasing this to P1

**Comment by bkonyi** [comment]
Just waiting on a LGTM for [this CL](https://dart-review.googlesource.com/c/sdk/+/164601).

**Comment by craggy2593** [comment]
Not sure this helps, and obviously there looks to already be some work already done to resolve, but was experiencing this earlier today when run from VSCode, and was being thrown at the same time I had a fatal error being reported.

The error I don't feel is related (was to do with calling the redundant `super.configureFlutterEngine(flutterEngine);` when registering plugins, registering our 3rd party plugins twice), but was getting two Flutter logs being spat out with the `Bad state` error, which have since disappeared since fixing the other error, and simulators/physical devices all run smoothly.
